server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -273,6 +301,40 @@ public static MetadataVersion latest() {
         return VALUES[VALUES.length - 1];
     }
 
+    public Optional<MetadataVersion> previous() {
+        int idx = this.ordinal();
+        if (idx > 2) {
+            return Optional.of(MetadataVersion.values()[idx - 1]);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    public static boolean checkIfMetadataChanged(MetadataVersion sourceVersion, MetadataVersion targetVersion) {
```

Comment:
Just clarifying, this is to check if the metadata has changed in an incompatible way between the sourceVersion and targetVersion?

Reply:
Let's add a test for this

Reply:
That's right, this will test if two versions have compatible metadata based on the `didMetadataChange` boolean added to the enum